com.unity.multiplayer.mlapi/Tests/Editor/ProfilerTests.cs

```diff
@@ -0,0 +1,192 @@
+        }
+
+        public void Send(string testMessage)
+        {
+            PerformanceDataManager.Increment(ProfilerConstants.TransportTestData);
+        }
+    }
+
+    public class TestHasProfilable : IHasProfilableTransport
+    {
+        internal static class ProfilerConstants
+        {
+            public const string NetworkTestData = nameof(NetworkTestData);
+        }
+
+        private TestTransport m_Transport;
+
+        public ITransportProfilerData GetTransport()
+        {
+            return m_Transport;
+        }
+
+        public void Initialize(bool useNullTransport)
+        {
+            m_Transport = useNullTransport ? null : new TestTransport();
+            ProfilerNotifier.Initialize(this);
+        }
+
+        public static void ProfilerBeginTick()
+        {
+            ProfilerNotifier.ProfilerBeginTick();
+        }
+
+        public static void NotifyProfilerListeners()
+        {
+            ProfilerNotifier.NotifyProfilerListeners();
+        }
+
+        public void Send()
+        {
+            ProfilerNotifier.Increment(ProfilerConstants.NetworkTestData);
+            m_Transport?.Send("testMessage");
+        }
+    }
+
+    public class NoTickDataException : Exception
+    {
+    }
+
+    public class ProfilerTests
+    {
+        private static void BreakDownTestProfiler(bool useNullTransport)
+        {
+            if (useNullTransport)
+            {
+                ProfilerNotifier.OnPerformanceDataEvent -= TestProfilerOnPerformanceDataEventNoTransport;
+            }
+            else
+            {
+                ProfilerNotifier.OnPerformanceDataEvent -= TestProfilerOnPerformanceDataEventNormal;
+            }
+
+            ProfilerNotifier.OnNoTickDataEvent -= TestProfilerNotifierOnOnNoTickDataEvent;
+        }
+
+        private static TestHasProfilable SetupTestProfiler(bool useNullTransport)
+        {
+            ProfilerNotifier.OnNoTickDataEvent += TestProfilerNotifierOnOnNoTickDataEvent;
+            if (useNullTransport)
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNoTransport;
+            }
+            else
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNormal;
+            }
+
+            EditorApplication.UnlockReloadAssemblies();
+            var testProfiler = new TestHasProfilable();
+            testProfiler.Initialize(useNullTransport);
+            return testProfiler;
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlowNull()
+        {
+            const bool useNullTransport = true;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            BreakDownTestProfiler(useNullTransport);
```

Comment:
You should use a [Teardown] method instead of the tailed method call here, otherwise an exception in a single test will leave it in a bad state